src/charm.py

```diff
@@ -99,18 +118,42 @@ def __init__(self, *args):
             self._on_loki_push_api_endpoint_departed,
         )
 
-    def update_alerts_rules(self, _):
+    def _metrics_alerts(self, event):
+        self.update_alerts_rules(
+            event,
+            alerts_func=self._scrape.alerts,
+            reload_func=self._remote_write.reload_alerts,
+            mapping=self.metrics_rules_paths,
+        )
+
+    def _loki_alerts(self, event):
+        self.update_alerts_rules(
+            event,
+            alerts_func=self._loki_provider.alerts,
+            reload_func=self._loki_consumer._reinitialize_alert_rules,
+            mapping=self.loki_rules_paths,
+        )
+
+    def update_alerts_rules(
+        self, _, alerts_func: Any, reload_func: Callable, mapping: RulesMapping
```

Comment:
Bit surprising to see a positional argument that is not being used. I presume it is because this function definition needs to confirm to some function signature.

Reply:
It's the actual event, which we ignore in a surprising amount of charms, it's just really obvious here.